File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.java
public class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int num : arr) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        Set<Integer> set = new HashSet<>();

        for (int count : map.values()) {
            if (set.contains(count)) {
                return false;
            }
            set.add(count);
        }

        return true;
    }
}

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        
        vector<int> ans;
        unordered_map<int, int> m;
        
        for(int i=0; i<arr.size(); i++)
        {
            m[arr[i]]++;
        }
        
        unordered_set<int> s;
        
        for(int i=0; i<arr.size(); i++)
        {
            s.insert(m[arr[i]]);
        }
        
        return (s.size()==m.size());
    }
};